helix-term/src/ui/editor.rs

```diff
@@ -117,8 +117,15 @@ impl EditorView {
             Self::highlight_cursorline(doc, view, surface, theme);
         }
 
-        let highlights = Self::doc_syntax_highlights(doc, view.offset, inner.height, theme);
-        let highlights = syntax::merge(highlights, Self::doc_diagnostics_highlights(doc, theme));
+        let mut highlights = Self::doc_syntax_highlights(doc, view.offset, inner.height, theme);
+        if editor.config().rainbow_brackets {
+            highlights = Box::new(syntax::merge_iters(
+                highlights,
+                Self::doc_rainbow_highlights(doc, view.offset, inner.height, theme),
+            ));
+        }
```

Comment:
I think it wouldn't overlay correctly: syntax highlights for `punctuation.bracket` would overlay the rainbow colors which would remove the rainbows in most themes.

I think it could still work to have selection highlights blend colors. Once rainbow highlights are merged onto syntax highlights, you can't really tell what's a syntax highlight or a rainbow highlight anymore from the render loop

Reply:
I think merging both together is still fine, but maybe we want to add some context as of what color is being applied?

I am thinking to add additional context, so later we have the ability style selection color differently based on the context, but I am not sure how well that is going to work.